server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -358,17 +360,31 @@ public DiscoveryNode(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
Keep it 2.17 now and backport or else we will have to switch it back after merging in 2.x

Reply:
Ack.